nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/AccessResource.java

```diff
@@ -592,6 +605,12 @@ public Response createUiExtensionToken(@Context HttpServletRequest httpServletRe
             throw new AccessDeniedException("No user authenticated in the request.");
         }
 
+        // authorize access
+        serviceFacade.authorizeAccess(lookup -> {
+            final Authorizable controller = lookup.getController();
+            controller.authorize(authorizer, RequestAction.READ, NiFiUserUtils.getNiFiUser());
```

Comment:
In retrospect I'm not sure this is the best way to handle authorizing this request. I will close this PR for now.